scripts/build.py

```diff
@@ -100,13 +125,107 @@ def copy_files_source_to_target(source, target):
+            if any(source_path.endswith(p) for p in FILE_EXTENSIONS_TO_IGNORE):
+                continue
+
+            target_path = source_path
+            relative_filepath = os.path.relpath(source_path, source)
+            if not relative_filepath.endswith(FILE_EXTENSIONS_NOT_TO_RENAME):
+                filepath, file_extension = os.path.splitext(relative_filepath)
+                relative_filepath = (filepath + '.' +
+                                     file_hashes[relative_filepath] +
+                                     file_extension)
+
+            target_path = os.path.join(os.getcwd(), target, relative_filepath)
             ensure_directory_exists(target_path)
             shutil.copyfile(source_path, target_path)
 
 
-def build_files(source, target):
+def is_file_hash_provided_to_frontend(filepath):
+    for folder in FOLDERS_PROVIDED_TO_FRONTEND:
+        if folder in filepath:
+            return True
+    for ending in FILE_ENDINGS_PROVIDED_TO_FRONTEND:
+        if filepath.endswith(ending):
+            return True
+    return False
+
+
+def generate_md5_hash(filepath):
+    """Returns md5 hash of file.
+
+    Args:
+        filepath: str. Path to the file.
+
+    Returns:
+        str. Hexadecimal hash of specified file.
+    """
+    m = hashlib.md5()
+    with open(filepath, 'rb') as f:
+        while True:
+            buf = f.read(HASH_BLOCK_SIZE)
+            if not buf:
+                break
+            m.update(buf)
+    return m.hexdigest()
+
+
+def get_file_hashes(folder_path):
+    """Returns hashes of all files in folder tree.
+
+    Args:
+        folder_path: str. Root folder of the tree.
+
+    Returns:
+        dict(str, str). Dictionary with keys specifying file paths and values
+        specifying file hashes.
+    """
+    file_hashes = dict()
+
+    print('Computing hashes for files in %s'
+          % os.path.join(os.getcwd(), folder_path))
+
+    for root, dirs, files in os.walk(os.path.join(os.getcwd(), folder_path)):
+        for directory in dirs:
+            print('Computing hashes for files in %s'
+                  % os.path.join(root, directory))
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            relative_filepath = os.path.relpath(filepath, folder_path)
+            file_hashes[relative_filepath] = generate_md5_hash(filepath)
+
+    return file_hashes
+
+
+def filter_hashes(file_hashes):
+    filtered_hashes = dict()
+    for filepath, file_hash in file_hashes.iteritems():
+        if is_file_hash_provided_to_frontend(filepath):
+            filtered_hashes['/' + filepath] = file_hash
+    return filtered_hashes
+
+
+def save_hashes_as_json(target_filepath, file_hashes):
+    # Only some of the hashes are needed in the frontend.
+    filtered_hashes = filter_hashes(file_hashes)
+
+    hashes_json = json.dumps(filtered_hashes)
+    with open(target_filepath, 'w') as f:
+        f.write('var hashes = ' + hashes_json + ';')
```

Comment:
Generally we do interpolation: `'var hashes = %s;' % (...)`

Also, flagging that this needs JSON.parse().